beps/bep_0003.rst

```diff
@@ -380,6 +429,24 @@ decent chance of getting a complete piece to upload, new connections
 are three times as likely to start as the current optimistic unchoke
 as anywhere else in the rotation.
 
+
+------------
+Upgrade Path
+------------
+
+## TODO ##
+
+* restrict file layout. no nested directories for hybrid torrents
+* padding or different piece-space layout?
+* pieces field
+* double announce behavior
+* safe hashing. avoid downgrade attacks
+* changes to BEP 9. magnets. send merkle layers.
```

Comment:
Yes, as I said in the initial comment, this is intentional to decouple hash transfer from pieces transfer. The tribler approach is great for one thing, terrible for everything else.